libs/serialization/src/exception_ptr.cpp

```diff
@@ -25,281 +25,316 @@
+            }
+
+            // figure out concrete underlying exception type
+            try
+            {
+                std::rethrow_exception(ep);
+            }
+            catch (hpx::thread_interrupted const&)
+            {
+                type = hpx::util::hpx_thread_interrupted_exception;
+                what = "hpx::thread_interrupted";
+                err_value = hpx::thread_cancelled;
+            }
+            catch (hpx::exception const& e)
+            {
+                type = hpx::util::hpx_exception;
+                what = e.what();
+                err_value = e.get_error();
+            }
+            catch (boost::system::system_error const& e)
+            {
+                type = hpx::util::boost_system_error;
+                what = e.what();
+                err_value = e.code().value();
+                err_message = e.code().message();
+            }
+            catch (std::runtime_error const& e)
+            {
+                type = hpx::util::std_runtime_error;
+                what = e.what();
+            }
+            catch (std::invalid_argument const& e)
+            {
+                type = hpx::util::std_invalid_argument;
+                what = e.what();
+            }
+            catch (std::out_of_range const& e)
+            {
+                type = hpx::util::std_out_of_range;
+                what = e.what();
+            }
+            catch (std::logic_error const& e)
+            {
+                type = hpx::util::std_logic_error;
+                what = e.what();
+            }
+            catch (std::bad_alloc const& e)
+            {
+                type = hpx::util::std_bad_alloc;
+                what = e.what();
+            }
+            catch (std::bad_cast const& e)
+            {
+                type = hpx::util::std_bad_cast;
+                what = e.what();
+            }
+            catch (std::bad_typeid const& e)
+            {
+                type = hpx::util::std_bad_typeid;
+                what = e.what();
+            }
+            catch (std::bad_exception const& e)
+            {
+                type = hpx::util::std_bad_exception;
+                what = e.what();
+            }
+            catch (std::exception const& e)
+            {
+                type = hpx::util::std_exception;
+                what = e.what();
+            }
+#if BOOST_ASIO_HAS_BOOST_THROW_EXCEPTION != 0
+            catch (boost::exception const& e)
+            {
+                type = hpx::util::boost_exception;
+                what = boost::diagnostic_information(e);
+            }
+#endif
+            catch (...)
+            {
+                type = hpx::util::unknown_exception;
+                what = "unknown exception";
+            }
 
-        static load_custom_exception_handler_type load_custom_exception_handler;
+            // clang-format off
+        ar & type & what & throw_function_ & throw_file_ & throw_line_;
```

Comment:
Is the indentation off or is it just GitHub playing games here?

Reply:
Uh, no, that's my editor. Thanks!